strawberryfields/gbs/similarity.py

```diff
@@ -201,3 +203,42 @@ def event_to_sample(photon_number: int, max_count_per_mode: int, modes: int) ->
             available_modes.remove(j)
 
     return sample
+
+
+def feature_vector_sampling(
+    samples: list, event_photon_numbers: list, max_count_per_mode: int = 2
+) -> np.ndarray:

```

Comment:
I'm still not convinced we want a numpy array as output. Josh mentioned previously it would make normalizing vectors easier, but I'm not sure that is actually relevant. Certainly the whole GBS layer has previously operated with lists, with this function being the only exception. For example, already during the tests below we need to convert back to list

Reply:
I'm ok to change the output to be a list. Was thinking about what impact this would have if we do a `gram_matrix` function and I think it should be fine. We could do something like
```python
vecs = [feature_vector_sampling(samples, [2, 4, 6, 8]) for samples in composite_samples]
gram = gram_matrix(vecs)
```

Reply:
As a side note, I assumed we were going with unnormalized feature vectors - is this the advice we got?

Reply:
@josh146 - on board to switch to list instead of numpy array?

Reply:
Yep, happy to switch back to a list. I admit it's slight personal bias --- I tend to prefer NumPy arrays being returned!